doc/devdocs/locks.rst

```diff
@@ -0,0 +1,128 @@
+.. currentmodule:: Base
+
+****************************************************
+Proper maintenance and care of multi-threading locks
+****************************************************
+
+The following strategies are used to ensure that the code is dead-lock free
+(generally by addressing the 4th Coffman condition: circular wait).
```

Comment:
FWIW, I don't have CS background and I've never heard of these before. However, in this case, the name is clear enough for google to return the right result.